crates/chia-consensus/src/gen/conditions.rs

```diff
@@ -99,14 +119,14 @@ impl SpendVisitor for MempoolVisitor {
             Condition::AggSigMe(_, _)
             | Condition::AggSigParent(_, _)
             | Condition::AggSigParentAmount(_, _)
-            | Condition::AggSigParentPuzzle(_, _) => {
+            | Condition::AggSigParentPuzzle(_, _)
+            | Condition::AggSigAmount(_, _)
+            | Condition::AggSigPuzzleAmount(_, _) => {
+                // references to your parent and references will not successfully fastforward
                 spend.flags &= !ELIGIBLE_FOR_DEDUP;
                 spend.flags &= !ELIGIBLE_FOR_FF;
             }
-            Condition::AggSigPuzzle(_, _)
-            | Condition::AggSigAmount(_, _)
-            | Condition::AggSigPuzzleAmount(_, _)
-            | Condition::AggSigUnsafe(_, _) => {
+            Condition::AggSigPuzzle(_, _) | Condition::AggSigUnsafe(_, _) => {
                 spend.flags &= !ELIGIBLE_FOR_DEDUP;
             }
             Condition::SendMessage(src_mode, _dst, _msg) => {
```

Comment:
I think you need to restrict sent and received messages that commit to coin amount as well

Reply:
Done